test/powershell/engine/Encoding.Tests.ps1

```diff
@@ -0,0 +1,173 @@
+Describe "Encoding classes and methods are available" -Tag CI {
+    BeforeAll {
+        $testString = "t" + ([char]233) + "st"
+        $provider = get-item $TESTDRIVE
+        $testFile = "${TESTDRIVE}/file.txt"
+        $preamble = @{
+            Ascii = ''
+            BigEndianUTF32 = '254-255'
+            BigEndianUnicode = '254-255'
+            Byte = '255-254'
+            Default = ''
+            Oem = ''
+            String = '255-254'
+            UTF32 = '255-254-0-0'
+            UTF7 = ''
+            UTF8 = ''
+            UTF8BOM = '239-187-191'
+            UTF8NoBOM = ''
+            Unicode = '255-254'
+            Unknown = ''
+            WindowsLegacy = ''
+        }
+
+        function Get-FileBytes
+        {
+            param ( $file, [int]$count = [int]::MaxValue )
+            (Get-Content $file -Encoding byte | Select-Object -First $count) -Join "-"
+        }
+
+        function Get-NewLineBytes
+        {
+            param ( [Microsoft.PowerShell.FileEncoding]$encoding )
+            $encoder = [Microsoft.PowerShell.PowerShellEncoding]::GetEncoding($encoding)
+            $encoder.GetBytes([Environment]::NewLine) -Join "-"
+        }
+        function Test-GetEncoding
+        {
+            [CmdletBinding()]
+            param (
+                [Microsoft.PowerShell.FileEncoding]$Encoding
+            )
+            END {
+                [Microsoft.PowerShell.PowerShellEncoding]::GetCmdletEncoding($pscmdlet, $encoding)
+            }
+        }
+
+        $preambleTests =
+            @{ Name = 'Ascii'; Preamble = '' },
```

Comment:
The values that have an empty Preamble mask possible bugs and unexpected preamble may be present but it won't be detected resulting in a false positive. For example, the test won't detect that a UTF32 encoding was encountered when ASCII was expected.
I suggest coding the test data to have a well known sequence following the BOM (or lack of BOM) and have the Preamble strings include this sequence. This will help reduce potential false positives.